src/NUnitFramework/nunitlite/TextRunner.cs

```diff
@@ -361,12 +363,52 @@ private int ExploreTests()
                 runSettings[FrameworkPackageSettings.DefaultTestNamePattern] = options.DefaultTestNamePattern;
 
             if (options.TestParameters != null)
-                runSettings[FrameworkPackageSettings.TestParameters] = options.TestParameters;
+            {
+                var dictionary = new Dictionary<string, string>();
```

Comment:
As with the console, I'd prefer to see the dictionary creation centralized in the Options hierarchy so that the framework assembly simply receives a dictionary. Otherwise, there is no point in making it a framework run setting. The framework isn't intended to change the settings that are passed to it, just to use them.

As an aside, I suppose that working on so many versions of the same thing is giving both you and the reviewers extra work. 😢